nixos/modules/virtualisation/oci-common.nix

```diff
@@ -0,0 +1,60 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.oci;
+in
+{
+  imports = [ ../profiles/qemu-guest.nix ];
+
+  # Taken from /proc/cmdline of Ubuntu 20.04.2 LTS on OCI
+  boot.kernelParams = [
+    "nvme.shutdown_timeout=10"
+    "nvme_core.shutdown_timeout=10"
+    "libiscsi.debug_libiscsi_eh=1"
+    "crash_kexec_post_notifiers"
+
+    # VNC console
+    "console=tty1"
+
+    # x86_64-linux
+    "console=ttyS0"
+
+    # aarch64-linux
+    "console=ttyAMA0,115200"
```

Comment:
Does it work having all the three? I know it works for aarch64 because it's what I used/use, but I didn't see this before.

Reply:
Yes: https://www.kernel.org/doc/html/latest/admin-guide/serial-console.html

> It is possible to specify multiple devices for console output. You can define a new kernel command line option to select which device(s) to use for console output.